pycode/memilio-epidata/memilio/epidata/modifyDataframeSeries.py

```diff
@@ -70,25 +70,57 @@ def impute_and_reduce_df(
         pass
 
     # range of dates which should be filled
-    if min_date == '':
-        min_date = min(df_old[dd.EngEng['date']])
-    if max_date == '':
-        max_date = max(df_old[dd.EngEng['date']])
+    # to prevent inconsistent data with different start dates, all dates are filled
+    # TODO: find a better method, to prevent unnecessary computation of dates before start date
+    first_date = min(df_old[dd.EngEng['date']])
+    last_date = max(df_old[dd.EngEng['date']])
+
+    # Transform dates to datetime
+    if isinstance(first_date, str) == True:
+        first_date = datetime.strptime(first_date, "%Y-%m-%d")
+    if isinstance(last_date, str) == True:
+        last_date = datetime.strptime(last_date, "%Y-%m-%d")
+
+    # Transform timestamp to date
+    try:
+        first_date = first_date.date()
+        last_date = last_date.date()
+    except:
+        pass
+    # range of dates which should be in output
+    if (min_date is None) or (min_date == ''):
+        min_date = first_date
+    if (max_date is None) or (max_date == ''):
+        max_date = last_date
 
     # Transform dates to datetime
     if isinstance(min_date, str) == True:
         min_date = datetime.strptime(min_date, "%Y-%m-%d")
     if isinstance(max_date, str) == True:
         max_date = datetime.strptime(max_date, "%Y-%m-%d")
 
+    try:
+        min_date = min_date.date()
+        max_date = max_date.date()
+    except:
+        pass
```

Comment:
I'm not sure on this one. This does transform Timestamp and datetime.datetime to datetime.date (which is important for the comparison of `first_date` and `min_date_to_use` in line 118). However, the dtypes of `min_date` and `max_date` can be the same as in the dataframe if no value is given (where it would be useful to move this under the if condition in line 91) or it could be a given value with a different format than datetime.date (here it should be in another if-else block e.g. line 97). To cover both cases I think it should be outside of the if-else block.